graylog2-server/src/test/java/org/graylog2/migrations/V20191219090834_AddSourcesPageTest.java

```diff
@@ -0,0 +1,145 @@
+                .revision(1)
+                .name("Sources Page")
+                .description("description")
+                .vendor("")
+                .url(URI.create("http://graylog.com"))
+                .entities(ImmutableSet.of())
+                .build();
+        final ContentPackInstallation contentPackInstallation = ContentPackInstallation.builder()
+                .contentPackId(contentPack.id())
+                .contentPackRevision(1)
+                .parameters(ImmutableMap.of())
+                .entities(ImmutableSet.of())
+                .comment("Comment")
+                .createdAt(Instant.now())
+                .createdBy("admin")
+                .build();
+        when(configService.get(V20191219090834_AddSourcesPage.MigrationCompleted.class)).thenReturn(null);
+        when(contentPackPersistenceService.insert(contentPack)).thenReturn(Optional.ofNullable(contentPack));
+        when(contentPackService.installContentPack(contentPack, Collections.emptyMap(), "Add Sources Page", "admin")).thenReturn(contentPackInstallation);
+        when(objectMapper.readValue(contentPackURL, ContentPack.class)).thenReturn(contentPack);
+        migration.upgrade();
+
+        verify(contentPackService).installContentPack(contentPack, Collections.emptyMap(), "Add Sources Page", "admin");
+        verify(configService).write(V20191219090834_AddSourcesPage.MigrationCompleted.create("04fcf179-49e0-4e8f-9c02-0ff13062efbe"));
+    }
+
+    @Test
+    public void upgradeFailesBecauseSourcePageContentPackExists() throws IOException {
+        final URL contentPackURL = V20191219090834_AddSourcesPage.class
+                .getResource("V20191219090834_AddSourcesPage_Content_Pack.json");
+        final ContentPack contentPack = ContentPackV1.builder()
+                .id(ModelId.of("04fcf179-49e0-4e8f-9c02-0ff13062efbe"))
+                .summary("summary")
+                .revision(1)
+                .name("Sources Page")
+                .description("description")
+                .vendor("")
+                .url(URI.create("http://graylog.com"))
+                .entities(ImmutableSet.of())
+                .build();
+        final ContentPackInstallation contentPackInstallation = ContentPackInstallation.builder()
+                .contentPackId(contentPack.id())
+                .contentPackRevision(1)
+                .parameters(ImmutableMap.of())
+                .entities(ImmutableSet.of())
+                .comment("Comment")
+                .createdAt(Instant.now())
+                .createdBy("admin")
+                .build();
+        when(configService.get(V20191219090834_AddSourcesPage.MigrationCompleted.class)).thenReturn(null);
+        when(objectMapper.readValue(contentPackURL, ContentPack.class)).thenReturn(contentPack);
+        when(contentPackPersistenceService.insert(contentPack)).thenReturn(Optional.empty());
+        boolean exceptionCalled = false;
+        try {
+            migration.upgrade();
```

Comment:
You can do `assertThatExceptionOfType(ContentPackException.class).isThrownBy(() -> migration.upgrade());` instead.